File: June-LeetCoding-Challenge/10-Search-Insert-Position/Search-Insert-Position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        return distance(nums.begin(), lower_bound(nums.begin(), nums.end(), target));
    }
};


class Solution2 {
public:
    int searchInsert(vector<int>& nums, int target) {
       	return lower_bound(nums.begin(), nums.end(), target) - nums.begin();
    }
};

File: June-LeetCoding-Challenge/10-Search-Insert-Position/Search-Insert-Position.kt
class SearchInsertPositionKotlin35 {
    fun searchInsert(nums: IntArray, target: Int): Int {
        if (nums.isEmpty()) {
            return -1
        }
        var left = 0
        var right = nums.size - 1
        while (left + 1 < right) {
            val mid = left + (right - left) / 2
            when {
                nums[mid] <= target -> left = mid
                nums[mid] > target -> right = mid
            }
        }
        return when {
            target <= nums[left] -> left
            target <= nums[right] -> right
            else -> right + 1
        }
    }
}